platform/onebox/backup_and_restore/registry/import.sh

```diff
@@ -0,0 +1,140 @@
+#!/bin/bash
+
+: '
+This script exports the images corresponding to currently available skills
+needs a json for creds and import directory
+[Script Version -> 21.4]
+'
+
+red=$(tput setaf 1)
+green=$(tput setaf 2)
+yellow=$(tput setaf 3)
+default=$(tput sgr0)
+
+readonly REGISTRY_IMPORT_FILE=$1
+readonly IMPORT_PATH=$2/registry
+
+# Validate dependecny module
+# $1 - Name of the dependecny module
+# $2 - Command to validate module
+function validate_dependency() {
+  list=$($2)
+  if [ -z "$list" ]; then
+    echo "$red $(date) Please install ******** $1 ***********  ... Exiting $default"
+    exit 1
+  fi
+}
+
+# Validate required modules exits in target setup
+function validate_setup() {
+  validate_dependency curl "curl --version"
+  validate_dependency jq "jq --version"
+  validate_dependency sqlcmd "sqlcmd -?"
+  validate_dependency gzip "gzip --version"
+  validate_dependency docker "docker --version"
+  echo "$(date) Successfully validated required dependecies"
+}
+
+
+function validate_file_path() {
+  if [ ! -f "$1" ]; then
+    echo "$red $(date) $1 file does not exist, Please check ... Exiting $default"
+    exit 1
+  fi
+}
+
+function validate_input() {
+
+  # Validate file path
+  validate_file_path $REGISTRY_IMPORT_FILE
+
+  readonly DB_CONN=$(cat $REGISTRY_IMPORT_FILE | jq -r 'select(.dbConnection != null) | .dbConnection')
+  readonly DB_NAME=$(cat $REGISTRY_IMPORT_FILE | jq -r 'select(.dbName != null) | .dbName')
+  readonly DB_USER=$(cat $REGISTRY_IMPORT_FILE | jq -r 'select(.dbUser != null) | .dbUser')
+  readonly DB_PASSWORD="$(cat $REGISTRY_IMPORT_FILE | jq -r 'select(.dbPassword != null) | .dbPassword')"
+  readonly REGISTRY_ENDPOINT=$(cat $REGISTRY_IMPORT_FILE | jq -r 'select(.registryEndpoint != null) | .registryEndpoint')
+  readonly REGISTRY_USER=$(cat $REGISTRY_IMPORT_FILE | jq -r 'select(.registryUser != null) | .registryUser')
+  readonly REGISTRY_PASSWORD=$(cat $REGISTRY_IMPORT_FILE | jq -r 'select(.registryPassword != null) | .registryPassword')
+  readonly OLD_REGISTRY_ENDPOINT=$(cat $REGISTRY_IMPORT_FILE | jq -r 'select(.oldRegistryEndpoint != null) | .oldRegistryEndpoint')
+  
+
+  if [[ -z $DB_CONN || -z $DB_NAME || -z $DB_USER || -z $DB_PASSWORD || -z TENANT_NAME || -z REGISTRY_ENDPOINT || -z REGISTRY_USER || -z REGISTRY_PASSWORD || -z EXPORT_PATH || -z OLD_REGISTRY_ENDPOINT ]]; then
+    echo "$red $(date) Input is invalid or missing, Please check ... Exiting $default"
+    exit 1
+  fi
+
+  echo "$green $(date) Successfully validated user input $default"
+}
+
+formulate_docker_command() {
+    docker images registry
+    if [[ $? -ne 0 ]]; then
+        echo "sudo permission required for docker"
+        DOCKER_COMMAND="sudo docker"
+    else
+        DOCKER_COMMAND="docker"
+    fi
+}
+
+function docker_setup() {
```

Comment:
why is this required and why are we not connecting to docker through user name and password

Reply:
existing scripts behave in similar manner, copied from there